### DISCUSSION TITLE
Reimplement "Native" titlebar and "native" controls

### DISCUSSION THREAD

**Discussion by ImSwordQueen** [BrowserWorks/waterfox#4102] [Ideas]
**Is your feature request related to a problem? Please describe.**
Mozilla removed all "native" controls from Firefox since 117 due to the removal of Windows 7/8.x support but some themes/userstyles on Windows 10+ benefited from having those native controls. 

As an example i compiled two screenshots: One from Firefox 115 ESR with native control patch and one from the latest version of Waterfox (As of me writing this issue) both having the theme Geckium installed.

![Image](https://github.com/user-attachments/assets/d3f52c48-31ba-4acb-b7b8-702d7d5926bc)

![Image](https://github.com/user-attachments/assets/03dda1fb-b68d-4735-ab13-3883f224e783)

**Describe the solution you'd like**
It would be appreciated if re-implementing the native controls that were removed when waterfox switched from the 115ESR base to 128ESR base could be brought back as an optional feature.
 
**Describe alternatives you've considered**
There was a different Firefox fork that had this feature but it would be cool to have waterfox to have this feature as well. Also, some of the re-implementation of native controls was merged with pull request #3294 and then removed when switched to the ESR 128 base. 

**Additional context**
As shown in the screenshots, Native controls would make it that certain themes that rely on native controls from 115 could be used on waterfox without compromises.

**Comment by AllieTheFox** [comment]
I think this is a good idea to reimplement this feature in Waterfox for the reason being that a lot of custom themes for Firefox these days are stuck with 2 options either being to stick with Firefox 115 ESR which is inevitably going to lose support one day, or the the worse option being to base it off a modern Firefox version but have the downside of not having native controls/titlebars which severely impacts the look of the theme if you are going for older Chrome/Firefox for example.

**Comment by ter298** [comment]
This was reimplemented in @K4sum1's project, r3dfox, if I'm not mistaken?

If there is no issue with potentially reusing/adapting (and of course attributing as needed) the code from another Firefox fork - even with different goals in mind - if they offer the same functionality that is desired, then there should be no problem, yes?

(Or just use the other fork, which I have noticed to be undergoing some sort of artificial boycott given it lacks coverage compared to other similar projects targeting older OS eg Supermium...)

**Comment by dominichayesferen** [comment]
It may be worthwhile to go ahead with it regardless, if you feel the desire to go with and maintain it, as the more the merrier, or so they say. That way, there's a Plan B for those who want Native Controls support if... anything happens to the former... that would end Native Controls.

**Comment by ImSwordQueen** [comment]
Bump after a year later?

**Comment by xodus2002** [comment]
Bump after a couple months later?

**Comment by ElastedAlorian** [comment]
Bump after half a year?
